bokeh/application/handlers/handler.py

```diff
@@ -20,6 +21,13 @@ def static_path(self):
         else:
             return self._static
 
+    def template(self):
+        """Returns an app-specific custom Jinja template, if applicable."""
+        if self.failed:
+            return None
+        else:
+            return self._template
```

Comment:
Could add debug logs for any template found, and a warning if it is being overwritten?